mintpy/bulk_plate_motion.py

```diff
@@ -0,0 +1,369 @@
+# Copyright (c) 2013, Zhang Yunjun, Heresh Fattahi         #
+# Author: Yuan-Kai Liu, May 2022                           #
+############################################################
+
+# Citation:
+# Oliver L. Stephenson, Yuan-Kai Liu, Zhang Yunjun, Mark Simons, and Paul Rosen. (2022)
+# The Impact of Plate Motions on Long-Wavelength InSAR-Derived Velocity Fields. Manuscript in preparation.
+
+# Extra dependency:
+#   + platemotion (https://github.com/lcx366/PlateTectonic)
+#   + astropy
+#   How to install both:
+#      option (1) pip install platemotion
+#      option (2) git clone git@github.com:lcx366/PlateTectonic.git $TOOL_DIR/PlateTectonic
+#                 echo 'export PYTHONPATH=$PYTHONPATH:$TOOL_DIR/PlateTectonic' >> ~/.bashrc
+#                 somehow install other dependencies in setup.py using your conda
+
+# To-Do List (updated 2022.5.30 Yuan-Kai Liu):
+#   + Replace scipy.interpolate with alternatives for efficiency. E.g.:
+#       skimage.resize https://scikit-image.org/docs/stable/auto_examples/transform/plot_rescale.html
+#       check mintpy usage https://github.com/insarlab/MintPy/blob/7adb3a11f875b832488a0c8e44c174d98b1df254/mintpy/tropo_gacos.py#L129
+#   + Calculate Euler rotation to multi-points ENU motion is slow (called by pmm2enu_at() here)
+#       In `platemotion` package, use array operation rather than for loops (at https://github.com/lcx366/PlateTectonic/blob/main/platemotion/classes/plate.py#L153)
+#   + Replace platemotion package by equations of Euler trasnformation to relax this dependency at all?
+
+import os
+import sys
+import scipy
+import pyproj
+import argparse
+import numpy as np
+
+from mintpy.utils import readfile, writefile, utils as ut
+from mintpy.mask import mask_matrix
+from mintpy.diff import diff_file
+from mintpy import reference_point
+from mintpy.save_gmt import get_geo_lat_lon
+from mintpy.solid_earth_tides import prepare_los_geometry
+
+try:
+    from platemotion import Plate
+    from astropy import units as u
+except ImportError:
+    msg = 'Can NOT import platemotion!'
+    msg += '\nCheck more details at https://github.com/lcx366/PlateTectonic.'
+    raise ImportError(msg)
+
+
+
+#########################################  Usage  ##############################################
+REFERENCE = """reference:
+  Stephenson, O. L., Liu, Y. K., Yunjun, Z., Simons, M., and Rosen, P., (2022), The Impact of Plate
+    Motions on Long-Wavelength InSAR-Derived Velocity Fields. [in preparation]
+"""
+EXAMPLE = """example:
+  bulk_plate_motion.py -g inputs/geometryGeo.h5                --om_sph  59.32 234.04 0.216   -m None
+  bulk_plate_motion.py -g inputs/geometryGeo.h5 -v velocity.h5 --om_sph  54.45 259.66 0.255   -m waterMask.h5
+  bulk_plate_motion.py -g inputs/geometryGeo.h5 -v velocity.h5 --om_cart 1.154 -0.136 1.444   -m maskTempCoh.h5
+  bulk_plate_motion.py -g inputs/geometryGeo.h5 -v velocity.h5 --enu     25.0 30.5 0.0        -m zero
+"""
```

Comment:
Two questions here for usage:

1. Should we use ONE AND ONLY ONE of `--om_sph`, `--om_cart` and `--enu` to specify the plate motion?

2. For the detailed values you used for each of the 3 options. Could you please a) add a comment above each command line, to specify for which plate these example values are? b) write a note/procedure on how to get these values, if one want to apply this correction to their own dataset?

Reply:
1. Yes, we use one and only one. I will further write that in the usage
2. I will do it now, sounds good!